openedx/openedx-platform
# feat: enable unit-level control over discussions by default [BD-38]

## Description

This PR changes the default behaviour of the discussions experience by making the previous "unit-level
visibility" the default mechanism for configuring discussions.

Prior to this PR, under the new discussions configuration experience, all units would automatically get
assigned a discussion topic and have discussions enabled for them (other than units in graded or exam
subsections). However, if authors wanted they could enable a custom visibility mode which would allow
toggling discussions on or off on a per-unit level.

This PR makes this custom visibility mode the standard behaviour (and eventually, only behaviour)
and enables discussion for all units by default. This replicates the behaviour that already existed,
however, now gives authors control over disabling discussions for individual units by default.

It also removes the ability to disable discussions for all units (while still keeping course-wide
discussions) enabled.

## Supporting information


## Testing instructions

- Once this PR is checked out, create a new course. 
- Create a few units in the course. 
- Clicking on Unit settings should show an "Enable discussion" checkbox 
    ![Screenshot 2022-08-29 at 12-02-24 Course Outline New Discussion Test Studio](https://user-images.githubusercontent.com/118837/187196777-cf9d94f5-71e4-402b-bce1-101aee893361.png)
- Publish the Units
- Check the API here: http://localhost:18000/api/discussion/v2/course_topics/<course_id> it should show the new units and a general course topic. The topic name should match the unit name, and they should have an "enabled_in_context" entry that's true.
- Disable discussion for one of the units. The Unit's corresponding topic should now have "enabled_in_context" be false. 
- Create another unit, but disable discussion before publishing it. 
- Now it should simply not appear in the API. 

## Deadline

None
